### PR TITLE
Add new project for mocking via DiskAsserter

### PR DESCRIPTION
**Pull request by mukut1994**

### CODE REVIEW

File: MK94.Assert.Mocking/Mocker.cs [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+                return;
+            }
+
+            EnsureExpectedOperationCalled(invocation, returnName + ".json");
+
+            var stepPath = Path.Combine(diskAsserter.PathResolver.GetStepPath(), returnName + ".json").Replace('\\', '/');
+
+            diskAsserter.Operations.Value.Add(new AssertOperation(OperationMode.Input, stepPath));
+
+            if (!MethodReturnIsVoid(invocation))
```
Comment (mithileshz) [inline MK94.Assert.Mocking/Mocker.cs]: swap it around and return early?

File: MK94.Assert.Mocking/Mocker.cs [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        /// <param name="actual">The implementation to use when running in write mode. <br />
+        /// The calls and results of this implementation will be record for future re-runs.</param>
+        /// <returns>A mocked instance of <typeparamref name="T"/></returns>
+        /// <param name="mocked">The result mock object</param>
+        /// 
+        /// <example>
+            /// <code>
+                /// 
+                /// DiskAssert.Default
+                ///     .Of&lt;T1&gt;(() => new ImplementationOfT(), out var mocked1)
```
Comment (mithileshz) [inline MK94.Assert.Mocking/Mocker.cs:59]: Is this correct?

File: MK94.Assert.NUnit.Test/MockTest.cs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+                .WithMocks()
+                .Of<IDatabase>(() => new Database(), out var database);
+
+            database.Insert(1, "Text 1");
+            database.Insert(2, "Text 2");
+            database.Insert(3, "Text 3");
+
+            DiskAssert.Matches("Step 1", database.Select(3));
+            DiskAssert.Matches("Step 2", database.Select(1));
+            DiskAssert.Matches("Step 3", database.Select(2));
```
Comment (mithileshz) [inline MK94.Assert.NUnit.Test/MockTest.cs:48]: Why are orders different?

File: MK94.Assert.Mocking/Mocker.cs (hunk 1 above)
Comment (mukut1994) [reply MK94.Assert.Mocking/Mocker.cs]: nice catch, thanks, fixed

File: MK94.Assert.Mocking/Mocker.cs (hunk 2 above)
Comment (mukut1994) [reply MK94.Assert.Mocking/Mocker.cs:59]: yea shows an example in rider, but not visual studio for some reason
image
tbh i'm not sure if i'm using it correctly

File: MK94.Assert.NUnit.Test/MockTest.cs (hunk 3 above)
Comment (mukut1994) [reply MK94.Assert.NUnit.Test/MockTest.cs:48]: Stress test, the mock shouldn't expect the read/write order to stay the same basically

**Review by mithileshz — APPROVED** [review]

**Pull request merged**